File: 剑指offer/剑指 Offer 10- II. 青蛙跳台阶问题.cpp
//动态规划

class Solution {
public:
    int numWays(int n) {
        if(n==0) return 1;
        if(n==1) return 1;
        int n1=1,n2=1,N;
        while(n>=2){
            N=(n1+n2)%(int(1e9+7));
            n2=n1;
            n1=N;
            n--;
        }
        return N;

    }
};

File: 剑指offer_java/剑指 Offer 10- II. 青蛙跳台阶问题.java
class Solution {
    public int numWays(int n) {
        if(n==0) return 1;
        if(n==1) return 1;
        int a=1, b=1,sum=0;
        while(n>1){
            n--;
            sum=(a+b)%1000000007;
            a=b;
            b=sum;
        }
        return sum;
    }
}
